### PR TITLE
BUG: Fix scalar methods to receive keyword arguments

### LINKED ISSUE #9512: Scalar astype method doesn't accept keyword arguments
According to [the docs](https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.scalars.html), numpy  scalars are supposed to have the same methods as arrays, but the `astype` method behaves differently by not accepting keyword arguments. This can cause issues for code that wants to accept scalars and arrays interchangeably e.g. see thunder-project/thunder#121 (not my use case, just the first hit I found).

Sample code (numpy 1.13.1):
```python
>>> import numpy as np
>>> np.float64(1).astype(np.float32, casting='same_kind')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: astype() takes no keyword arguments
```

### PR DESCRIPTION
**Pull request by Licht-T**
This fixes https://github.com/numpy/numpy/issues/9512 .

### CODE REVIEW

**Comment by eric-wieser** [comment]
LGTM, although I wonder if there are others with this problem, and we should just use the kwargs path for all of them - which at worst is a (small) performance hit.

**Comment by Licht-T** [comment]
@eric-wieser Thanks for your reply.
Your opinion is right in terms of performance.
I have no idea for avoiding a performance hit.

**Comment by eric-wieser** [comment]
Either way, this patch probably is good enough as is - just trying to start a discussion about future work

File: numpy/core/src/multiarray/scalartypes.c.src [hunk 1]
```diff
@@ -1529,7 +1529,7 @@ gentype_wraparray(PyObject *NPY_UNUSED(scalar), PyObject *args)
  */
 /**begin repeat
  *
- * #name = tolist, item, tostring, tobytes, astype, copy, __deepcopy__,
+ * #name = tolist, item, tostring, tobytes, copy, __deepcopy__,
  *         searchsorted, view, swapaxes, conj, conjugate, nonzero, flatten,
```
Comment (eric-wieser) [inline numpy/core/src/multiarray/scalartypes.c.src]: From `methods.c`, these also don't belong here:

* `tostring`
* `tobytes`
* `copy`
* `searchsorted`
* `view`
* `flatten`
* `ravel`

Comment (Licht-T) [inline numpy/core/src/multiarray/scalartypes.c.src]: Okay, I'll fix.

**Comment by Licht-T** [comment]
@eric-wieser Fixed.
Should I change the PR title?

**Comment by eric-wieser** [comment]
Wow, you didn't need to do each of those in a separate commit - that's just making work for yourself!

We can change PR titles easily from the web interface anyway, so don't worry about it - it's commit messages that are more work to change, but yours are fine already

**Comment by eric-wieser** [comment]
Separate commits did make it super-easy to review though, so thanks!

**Pull request merged**